carla-carma-integration/carla_carma_bridge/src/carla_carma_bridge/carma_carla_guidance

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+# Copyright (C) 2021 LEIDOS.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import rospy
+from cav_msgs.msg import RouteEvent
+from cav_msgs.msg import Plugin
+from cav_srvs.srv import SetGuidanceActive
+
+route_selected = False
+plugins_activated = False
+
+def check_route_status(route_event_msg):
+    global route_selected
+    if route_event_msg.event == 1:
+        print("Route Selected")
+        route_selected = True
+
+def check_plugin_status(plugin_msg):
+    global plugins_activated
+    if plugin_msg.activated == True:
+        print("Plugins Activated")
+        #NOTE: May have to add logic to parse through every individual plugin and determine their status
+        plugins_activated = True
+
+def initialize():
+    rospy.init_node("carma_carla_guidance")
+    service = rospy.ServiceProxy('/set_active_guidance', SetGuidanceActive)
+    #While-loop monitors the route and plugin status and sets guidance to active once both are confirmed true
+    
+    route_sub = rospy.Subscriber("/route_event",RouteEvent(), check_route_status)
+    plugin_sub = rospy.Subscriber("/plugin_discovery", Plugin(), check_plugin_status)
+    while rospy.is_shutdown() != True:  
```

Comment:
I think you need a spinOnce at the bottom of this while loop, otherwise you're not processing ROS messages at all and just doing a really tight loop here over variables that won't change.

Reply:
[It seems like there's no spinOnce command for rospy. ](https://answers.ros.org/question/110336/python-spin-once-equivalent/)